### PR TITLE
Seed SeLion 2.0.0-SNAPSHOT with Selenium 3-beta

### PR DESCRIPTION
**Pull request by mach6**
- Update SeLion grid to be Selenium 3-beta compatible
- Update Java version to 8
- Update Maven version to 3.2.x
- Add .editorconfig rules
- Add github new issue and PR templates
- Update REAEME.MD

### CODE REVIEW

**Comment by mach6** [comment]
Merging will close #310

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                new JCommander(new SeLionGridNodeConfiguration()).usage();
+            }
+        }).build();
+
+    private static void logEnvironment() {
+        LOGGER.fine("Environment Variables: " + Collections.singletonList(System.getenv()));
+        LOGGER.fine("JVM System Properties: " + Collections.singletonList(System.getProperties()));
+    }
+
+    public static void main(String[] args) throws Exception {
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        } else {
+            printInfoAboutRoles(role);
+            return;
+        }
+    }
+
+    public static void printInfoAboutRoles(String roleCommandLineArg) {
+        CliUtils.printWrappedLine("", "Error: the role '" + roleCommandLineArg
+                    + "' does not match a recognized server role\n");
+        System.out.println("Selenium server can run in one of the following roles:\n"
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    }
+
+    /**
+     * Boot the instance base on the arguments supplied
+     *
+     * @param args
+     *            the arguments to use
+     * @throws Exception
+     */
+    public void boot(String[] args) throws Exception {
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                    "Selenium build info: version: '%s', revision: '%s'",
+                    buildInfo.getReleaseLabel(),
+                    buildInfo.getBuildRevision()));
+
+            try {
+                launcher.launch();
+                type = launcher.type;
+            } catch (Exception e) {
+                launcher.printUsage();
+                e.printStackTrace();
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+                server.boot();
+                LOGGER.info("Selenium Server is up and running");
+            }
+
+            @Override
+            void printUsage() {
+                new JCommander(new SeLionStandaloneConfiguration()).usage();
+            }
+        }).put(GridRole.HUB, new SeLionGridItemLauncher() {
+            public void setConfiguration(String[] args) {
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+public class SeLionGridLauncherV3 {
+
+    private static final Logger LOGGER = SeLionGridLogger.getLogger(SeLionGridLauncherV3.class);
+    private Object type;
+
+    private static abstract class SeLionGridItemLauncher {
+        protected StandaloneConfiguration configuration;
+        protected boolean helpRequested;
+        protected Object type;
+        abstract void setConfiguration(String[] args);
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/servlets/ProxyInfo.java [hunk 7]
```diff
@@ -96,7 +97,7 @@
     /**
      * @see RemoteProxy#getConfig()
      */
-    private Map<String, Object> configuration;
+    private GridNodeConfiguration configuration;
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/servlets/ProxyInfo.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 2 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 1 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 3 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 4 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 5 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java (hunk 6 above)
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncherV3.java]: will not fix

**Review by sebady — APPROVED** [review]
LGTM

**Comment by mach6** [comment]
@sebady thanks.. I'll merge this then. fyi, there are some things about this commit I don't like but I'm already addressing them in my next PR. :)

**Pull request merged**